### DISCUSSION TITLE
When - Automatically clean up files left behind by interrupted uploads?

### DISCUSSION THREAD

**Discussion by chriswa81** [immich-app/immich#14544] [General]
I have a question about the new function “_Automatic deletion of failed upload files_”. When is this data deleted? I have tested it and interrupted various uploads. The data remains in the upload folder. 
I used to be able to delete them manually. This is canceled with an error message when trying to delete manually. Before the update, I was able to delete files manually from the upload folder.

**Comment by bo0tzz** [comment]
The way it works is that when an upload fails, the data that was being uploaded should get deleted. Other files that were already there from before this update aren't affected, you'll still need to clean those up manually.

**Reply by chriswa81** [reply]
Then the automatic cleanup does not work for me. Current upload aborts are not deleted.

**Reply by bo0tzz** [reply]
Are you on v1.122.1?

**Reply by chriswa81** [reply]
Yes.

**Comment by Acidiphilium** [comment]
Is there a way to find the corrupted uploaded files which were uploaded in the past by an interrupted uploads? How do we clean this upload folders? I remember in a previous immich version, it had a function to list all failed uploads but you removed that option so now how do we know what could be those failed videos in that upload folder?

Thanks for the great work again. I love immich really :-)

**Reply by alextran1502** [reply]
You can turn on `Storage Template` feature, then run `Storage Migration Job`, after that, the left over files in `upload` folder are the partial files

**Reply by Acidiphilium** [reply]
Thanks for the tips! May I ask you: can I then revert back from that operation without any trouble?

**Comment by chriswa81** [comment]
Was it possible to reproduce the issue or should I officially document it again here under "Issues"?

**Comment by alextran1502** [comment]
~~The data is deleted as soon as the upload is canceled. You can go to the Adminstration settings > Server Settings > set logs to verbose to see the actual file path that get removed~~

Ah wait, I only have that logged during development. How did you identify the partial files are not removed?

**Reply by chriswa81** [reply]
The failed upload data is still in the immich upload folder. It is definitely this data and can be reproduced.

**Reply by alextran1502** [reply]
Please note that this mechanism only applied for new upload, not removing existing failed upload files from previous version

**Reply by chriswa81** [reply]
I know. I am aware of that. That's why I can reproduce it with current upload tests with the current version 1.122.1.

**Reply by alextran1502** [reply]
Can you walk me through how do you identify the file isn't get deleted?

**Comment by chriswa81** [comment]
Here is my explanation of how I can determine that the incorrect uploads are not deleted:

My Upload-Location is:
![01](https://github.com/user-attachments/assets/a20cfe9d-6aa0-49f0-a482-53c869bfddbc)
![02](https://github.com/user-attachments/assets/90b55d47-44fd-4227-8e51-2114214a5778)

The interrupted uploads are then stored here:
![03](https://github.com/user-attachments/assets/f573e476-9e8f-4851-a0b1-827ee01c6f39)
This folder contains the interrupted data that is not deleted by immich.

**Reply by bo0tzz** [reply]
The folder itself is expected to stay around. What does it actually contain? If you reproduce the issue, do you see a _new_ file appear in there and stay around?

**Reply by chriswa81** [reply]
Correct uploads are uploaded or saved according to the storage template. Only canceled or incorrectly uploaded media remain in the upload folder.
If I reproduce it, a new folder is created (e.g. folder with name 6e) and a file with the size until the upload was canceled is stored in it.
These folders or data are not deleted, as it should happen according to the function of version 1.122.0.

**Reply by alextran1502** [reply]
For clarification, the file will be deleted, not the folder containing the file

**Reply by chriswa81** [reply]
And that is precisely not the case. The partial data is still available in the respective folders.

**Reply by alextran1502** [reply]
Let me try to double check to see if I can reproduce this

**Comment by adamzvolanek** [comment]
Currently sitting with thousands of sub-directories under the upload folder,

<img width="288" height="31" alt="image" src="https://github.com/user-attachments/assets/1f374fbf-7a49-4930-b9e4-832199f75a68" />


Is there a function or feature that helps clean this up? I don't think I have this many failed uploads, but if I am the cleanup is not happening. 

Running Immich v2.0.1
